src/main/java/org/apache/sysml/hops/codegen/opt/PlanSelectionFuseCostBasedV2.java

```diff
@@ -215,20 +215,23 @@ private void selectPlans(CPlanMemoTable memo, PlanPartition part)
 	 * @return optimal assignment of materialization points
 	 */
 	private static boolean[] enumPlans(CPlanMemoTable memo, PlanPartition part, StaticCosts costs, 
-		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC) 
+		ReachabilityGraph rgraph, InterestingPoint[] matPoints, int off, double bestC)
 	{
 		//scan linearized search space, w/ skips for branch and bound pruning
 		//and structural pruning (where we solve conditionally independent problems)
 		//bestC is monotonically non-increasing and serves as the upper bound
-		long len = (long)Math.pow(2, matPoints.length-off);
+		long len = 1L << matPoints.length-off;
```

Comment:
I doubt that this is an actual bottleneck and the Java JIT compiler likely does this strength reduction automatically. Hence, I would recommend to stick to the more readable form (unless you where able to measure a statistically significant difference).